src/app/modules/submission/multirespondent-submission/multirespondent-submission.controller.ts

```diff
@@ -347,6 +377,115 @@ const runMultirespondentWorkflow = ({
   )
 }
 
+const sendMrfCompletionEmailIfWorkflowCompleted = ({
+  currentStepNumber,
+  formWorkflow,
+  destinationEmails,
+  formId,
+  formTitle,
+  submissionId,
+}: {
+  currentStepNumber: number
+  formWorkflow: FormWorkflowDto
+  destinationEmails: string[]
+  formId: string
+  formTitle: string
+  submissionId: string
+}): ResultAsync<true, MailSendError> => {
+  const logMeta = {
+    action: 'sendMrfCompletionEmail',
+    formId,
+    submissionId,
+  }
+
+  const lastStepNumber = formWorkflow.length - 1
+  const isLastStep = currentStepNumber === lastStepNumber
+  const isWorkflowCompleted = isLastStep
+
+  if (isWorkflowCompleted) {
+    return MailService.sendMrfWorkflowCompletionEmail({
+      emails: destinationEmails,
+      formId,
+      formTitle,
+      responseId: submissionId,
+    }).orElse((error) => {
+      logger.error({
+        message: 'Failed to send workflow completion email',
+        meta: { ...logMeta, destinationEmails },
+        error,
+      })
+      return errAsync(error)
+    })
+  } else {
+    return okAsync(true)
+  }
+}
+
+const sendMrfOutcomeEmails = ({
```

Comment:
This function could likely be moved into `.service` as it doesn't look like it's dealing with `HttpRequests/Responses`